src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java

```diff
@@ -104,11 +99,16 @@ Optional<ReplicateTaskSummary> getAvailableReplicateTaskSummary(long workerLastB
 
         // TODO : Remove this, the optional can never be empty
```

Comment:
This TODO can be removed. It was relevant before the optimization.
Now `worker` becomes an argument from `canAcceptMoreWorks`.

Reply:
Good catch! Please see https://github.com/iExecBlockchainComputing/iexec-core/pull/644/commits/d0c0db7fd1a2a61d5135ed50783f50e61aecefc1.